packages/ui/src/contexts/MultiProxyContext.tsx

```diff
@@ -64,136 +76,116 @@ const MultiProxyContextProvider = ({ children }: MultisigContextProps) => {
       data.accounts.forEach((account) => {
-        // if the account is a pure proxy
-        if (account.isPureProxy) {
-          // find the delegatee that are multisigs and put all the infos right away
-          account.delegatorFor.forEach(({ delegatee, type }) => {
-            if (delegatee.isMultisig) {
-              const previousMultisigsForProxy = pureProxyMap.get(account.address)?.multisigs || []
-
-              const isAlreadyInMultisigList = !!previousMultisigsForProxy.find(
-                ({ address }) => address === delegatee.address
-              )
-
-              // do not add a second time a multisig
-              if (isAlreadyInMultisigList) return
-
-              const newMultisigForProxy = {
-                address: delegatee.address,
-                signatories: getSignatoriesFromAccount(delegatee.signatories),
-                threshold: delegatee?.threshold || undefined,
-                type
-              }
-
-              pureProxyMap.set(account.address, {
-                multisigs: [...previousMultisigsForProxy, newMultisigForProxy]
-              })
-            }
-          })
-
-          return
-        }
-
-        // from this point, we should only be dealing with multisigs
-        // looking for multisigs being delegatee for pure proxies
-        const pureProxyAddresses: {
-          pureAddress: string
-          type: ProxyType
-        }[] = []
-
-        // one multisig could be a delegator for multiple
-        account.isMultisig &&
-          account.delegateeFor.forEach(({ delegator, type }) => {
-            // if a pure was already added, e.g because it is watched
-            // we shouldn't associate this multisig to it twice
-            const currentMultisigsForProxy = pureProxyMap
-              .get(delegator.address)
-              ?.multisigs.map(({ address }) => address)
-
-            // finding all the accounts that are pure proxy and that don't include this multisig already
-            if (delegator?.isPureProxy && !currentMultisigsForProxy?.includes(account.address)) {
-              pureProxyAddresses.push({
-                pureAddress: delegator.address,
-                type: type
-              })
-            }
-          })
+        // find the delegatee that are multisigs and put all the infos right away
+        account.delegatorFor.forEach(({ delegatee, type }) => {
+          if (delegatee.isMultisig) {
+            const previousMultisigsForProxy = pureProxyMap.get(account.address)?.multisigs || []
 
-        // if this account is a multisig and is the delegatee for at least a pureProxy
-        if (account.isMultisig && pureProxyAddresses?.length > 0) {
-          pureProxyAddresses.forEach(({ pureAddress, type }) => {
-            const previousMultisigsForProxy = pureProxyMap.get(pureAddress)?.multisigs || []
+            const isAlreadyInMultisigList = !!previousMultisigsForProxy.find(
```

Comment:
```suggestion
            const isAlreadyInMultisigList = previousMultisigsForProxy.some(
```
More optimized as it returns immediately first time condition is true^^